content/en/docs/code/cri-api-dev-policies.md

```diff
@@ -0,0 +1,126 @@
+with the Kubernetes community, or one of existing container runtimes will stop
+engagement, this document will be updated.
+
+## Same maturity level (for beta and GA)
+
+Implementation of an API needed for a Kubernetes feature in a container runtime
+**must** be at least the same maturity as in k8s at a moment of Kubernetes release.
+This is similar to the [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior)
+when one feature is replacing another.
+
+With containerd and CRI-O it means that for the GA features, there should be a
+release of containerd and CRI-O implementing APIs needed for the feature in
+those container runtimes. And those APIs **must not** be marked as experimental
+features ([containerd experimental features](https://containerd.io/releases/#experimental-features)).
+For the beta, neither container runtime has a notion of beta feature or release and
+realistically the same maturity criteria applies as for GA.
+
+## Same maturity level (for alpha)
+
+There should be at least one implementation of an API needed for the Kubernetes
+feature merged into the container runtime default branch (main) or marked as
+experimental. An e2e test may demonstrate that the feature is working should be
+merged alongside the code. Note, tests may run against the default branch of a
+container runtime and the feature can be still not shipped.
+
+The actual container runtime release may be delayed to the later stage, but
+Kubernetes highly encourages fast availability of a release of a container
+runtime that can be tested by early adopters.
+
+## Minimal number of implementations
+
+Both containerd and CRI-O **must** have a GA release with the implementations of an
+API needed for a Kubernetes feature before this Kubernetes feature can be
+promoted to GA.
+
+## Safe Kubernetes defaults
+
+The feature cannot be enabled by default in Kubernetes as a beta feature before
+the required APIs are implemented in both container runtimes (containerd and
+CRI-O) and there is a GA release of a container runtime for each. The feature
+can be marked as beta, but disabled by default, if there is only one container
+runtime implementation of a required API that is released as GA. Note, as for
+any Kubernetes features, at least one release with the beta feature enabled by
+default is required before it is progressing to GA.
+
+## Guarantee portability
+
+The feature design (KEP PR) MUST be lgtm-ed by container runtime maintainers of
+CRI-O and containerd.
+
+The feature can only be merged as alpha in Kubernetes, if there is an agreement
+from both container runtime maintainers on the feature design in general and API
+shape. Gaining this agreement will often involve authoring the pull request
+demonstrating an API implementation to the both container runtime repositories
+or an alternative way for container runtime maintainers to confirm viability of
+suggested CRI APIs.
```

Comment:
Hm, would it be enough to state that the CRI API change has outlined in detail in the KEP? I'm not sure what the benefit is of bringing the API into the projects without the actual functionality.

Reply:
we need an agreement from runtime that this API is viable. Otherwise we may end up with the CRI API that is not implemented by any or one of runtimes. Getting review from two runtimes gives some confidence that it will be viable for other possible runtimes and will be portable enough